### PR TITLE
chore: Switch to Std.HashMap

### PR DESCRIPTION
**Pull request by bollu**
### Description:

This fixes deprecation warnings telling us to switch to `Std.HashMap

### Testing:

What tests have been run? Did `make all` succeed for your changes? Was
conformance testing successful on an Aarch64 machine?

### License:

By submitting this pull request, I confirm that my contribution is
made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: Tests/Tactics/CSE.lean [hunk 1]
```diff
@@ -269,80 +269,81 @@ hx2 : x9 + x3 = x2
...
 ⊢ x2 ++
-      ((x1 &&& x40 ^^^ ~~~x1 &&& x38) + (x1.rotateRight 14 ^^^ x1.rotateRight 18 ^^^ x1.rotateRight 41) +
+      ((x1 &&& x43 ^^^ ~~~x1 &&& x36) + (x1.rotateRight 14 ^^^ x1.rotateRight 18 ^^^ x1.rotateRight 41) +
         BitVec.extractLsb 63 0 x4) =
     x6 ++
-      (x36 + (x5.rotateRight 14 ^^^ x5.rotateRight 18 ^^^ x5.rotateRight 41) + (x5 &&& x40 ^^^ ~~~x5 &&& x38) + x35 +
+      (x37 + (x5.rotateRight 14 ^^^ x5.rotateRight 18 ^^^ x5.rotateRight 41) + (x5 &&& x43 ^^^ ~~~x5 &&& x36) + x41 +
         x44)
 -/
 #guard_msgs in theorem sha512h_rule_1 (a b c d e : BitVec 128) :
```
Comment (pennyannn) [inline Tests/Tactics/CSE.lean:349]: I wonder if this subgoal guard_msgs needs to be recorded. It seems very fragile.

File: Tests/Tactics/CSE.lean [hunk 2]
```diff
@@ -414,62 +415,63 @@ x20 x21 : BitVec 64
...
-      ((x1 &&& x37 ^^^ ~~~x1 &&& x42) + (x1.rotateRight 14 ^^^ x1.rotateRight 18 ^^^ x1.rotateRight 41) +
+      ((x1 &&& x37 ^^^ ~~~x1 &&& x36) + (x1.rotateRight 14 ^^^ x1.rotateRight 18 ^^^ x1.rotateRight 41) +
         BitVec.extractLsb 63 0 x4) =
     x11 ++
-      (x43 + (x5.rotateRight 14 ^^^ x5.rotateRight 18 ^^^ x5.rotateRight 41) + (x5 &&& x37 ^^^ ~~~x5 &&& x42) + x35 +
-        x36)
+      (x40 + (x5.rotateRight 14 ^^^ x5.rotateRight 18 ^^^ x5.rotateRight 41) + (x5 &&& x37 ^^^ ~~~x5 &&& x36) + x41 +
+        x39)
 -/
 #guard_msgs in theorem sha512h_rule_2 (a b c d e : BitVec 128) :
```
Comment (pennyannn) [inline Tests/Tactics/CSE.lean:476]: And this one as well.

File: Tests/Tactics/CSE.lean (hunk 1 above)
Comment (bollu) [reply Tests/Tactics/CSE.lean:349]: @pennyannn Yeah, I'm not particularly happy about the state of affairs either :( I'm keeping it around in hopes that we will eventually get regex matching on `guard_msgs`.

**Review by shigoel — APPROVED** [review]

File: Tests/Tactics/CSE.lean (hunk 1 above)
Comment (shigoel) [reply Tests/Tactics/CSE.lean:349]: Yes, I'm also expecting this to be stopgap, one way or the other :-) Maybe `cse` can elide var names when printing, or some such.

**Pull request merged**